### ISSUE TITLE
weight_decay doesn't have effect

### ISSUE THREAD

**Issue by kudkudak** [aigamedev/scikit-neuralnetwork#197]
I am not sure if I am using `Classifier` correctly, but following code returns same results for any `weight_decay` value. `weight_decay=1e5` should draw all weights to 0.

```{python}
import random
from sklearn import datasets
from sklearn.metrics import accuracy_score, mean_squared_error
from sknn.mlp import Classifier, Layer, NeuralNetwork

iris = datasets.load_iris()
for decay in [0, 1e5]:
    model = Classifier(
        layers=[
            Layer("Softmax")],
        regularize="L2",
        random_state=777,
        weight_decay=decay)
    model.fit(iris.data, iris.target)
    print(accuracy_score(model.predict(iris.data), iris.target))
```

Outputs:
0.953333333333
0.953333333333

**Comment by alexjc** [comment]
I recently went over this code and it seemed to work, but your output does look suspicious.

Can you compare the result of `get_parameters()` for each run? If it's identical then it's a bug. `1e5` seems very high, but I guess that's on purpose for testing?

**Comment by kudkudak** [comment]
@alexjc 

Yes it is on purpose :) Weights are the same, here is full log https://gist.github.com/kudkudak/d8ca12fec0c29572804c36875d9984fb

**Comment by pGit1** [comment]
I also notice dropout has little to no effect as well but I haven't done this specific experiment.

**Comment by alexjc** [comment]
Both `dropout` and `weight_decay` work fine in the latest version, did you upgrade? It was briefly a problem in 0.6 I think...

**Comment by kudkudak** [comment]
@alexjc Yes it was 0.6.1 (installed from pip, probably few weeks ago).  It works for 0.7 :) Thanks 👍

**Comment by alexjc** [comment]
I should add a version to the output log! Thanks for reporting back.

**Comment by pGit1** [comment]
Im on version 0.7.

**Comment by alexjc** [comment]
@pGit1 Looks fine to me if you have a specific bug report open a new ticket.
